src/Views/Wallpaper.vala

```diff
@@ -206,6 +206,7 @@ public class PantheonShell.Wallpaper : Gtk.Box {
         }
 
         gnome_background_settings.set_string ("picture-uri", uri);
+        gnome_background_settings.set_string ("picture-uri-dark", uri);
```

Comment:
You should reset the wallpaper uri instead of duplicating it to -dark key. Gala handles invalid path and fallbacks to light wallpaper.

```suggestion
        gnome_background_settings.set_string ("picture-uri-dark", "");
```